url2a908001cd_malictus_klang_tgz-pJ8-malictus_klang_test_PrimitiveTestJ8/src/malictus/klang/chunk/PlainTextChunk.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * This file is distributed under a BSD-style license. See the included LICENSE.txt file
+ * for more information. 
+ * Copyright (c) 2009, James Halliday
+ * All rights reserved.
+ */
+package malictus.klang.chunk;
+
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.charset.*;
+import malictus.klang.file.*;
+import malictus.klang.*;
+import malictus.klang.primitives.*;
+
+/**
+ * A class that represents any chunk that has only plain text in it, but that can be encoded in any different text encoding.
+ * @author Jim Halliday
+ */
+public class PlainTextChunk extends Chunk {
+	
+	String textEncoding = "";
+	
+	public PlainTextChunk(KlangFile parentFile, long startpos, long endpos, long datastartpos, long dataendpos, int chunkType, String chunkName, ContainerChunk parentChunk, RandomAccessFile raf, String textEncoding) throws IOException {
+		super(parentFile, startpos, endpos, datastartpos, dataendpos, chunkType, chunkName, parentChunk, raf);
+		this.textEncoding = textEncoding;
+		raf.seek(datastartpos);
+		/*
+		 * LOAD PRIMITIVE OBJECTS (UNINITIALIZED)
+		 */
+		/*
+		 * READ FROM FILE TO FILL PRIMITIVE DATA
+		 */
+		PlainString val  = null;
+		try {
+			long bytepos = raf.getFilePointer();
+			val = new PlainString(raf, dataendpos, textEncoding, true);
+			PrimitiveData valData = new PrimitiveData(val, KlangConstants.PRIMITIVE_DATA_PLAINSTRING_CHUNK_VALUE, bytepos, raf.getFilePointer());
+			/*
+			 * ADD PRIMITIVES TO VECTOR
+			 */
+			this.primitives.add(valData);
+		} catch (BadValueException er) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 47.